### ISSUE TITLE
[Bug]: Default value generator class doesn't work for Fieldcollections

### ISSUE THREAD

**Issue by MyZik** [pimcore/pimcore#11689]
### Expected behavior

If we add a field collection to our object and one of FC-fields has a default value generator class, this generator should define the value in FC-field.

### Actual behavior

By adding/saving a fieldcollection in object, default value will not be set. My class:

```php
<?php 

...

class TestDefaultValueGenerator implements DefaultValueGeneratorInterface
{
    /**
     * @param  Concrete  $object
     * @param  Data      $fieldDefinition
     * @param  array     $context
     *
     * @return string
     * @throws \Exception
     */
    public function getValue(Concrete $object, Data $fieldDefinition, array $context): string
    {
        return 'test';
    }
}
```

### Steps to reproduce

1. Create a default value generator class
2. Create a field collection
3. Add some fields to your field collection and set to one of them the default generator class
![Bildschirmfoto 2022-03-16 um 16 57 54](https://user-images.githubusercontent.com/12817320/158632997-d0cc8837-67ab-40d0-aa3a-454e581b8280.png)
4. Added your created field collection definition to your class
5. Create an object and try to add your field collection data.

**Comment by dvesh3** [comment]
duplicate of #9078

**Comment by mcop1** [comment]
> duplicate of #9078

Closed in favor of #9078
